src/library_fs.js

```diff
@@ -1135,12 +1134,11 @@ mergeInto(LibraryManager.library, {
       }
       if (stream.flags & {{{ cDefine('O_APPEND') }}}) {
         // seek to the end before writing in append mode
-        position = FS.llseek(stream, 0, {{{ cDefine('SEEK_END') }}});
```

Comment:
I didn't notice this the first time, but now it seems odd to me. don't we need to use llseek's return value?

Reply:
llseek itself sets `stream.position`, and we aleady uses stream.position when we detect non-seeking mode, so not really needed.

Reply:
Got it, thanks.

Reply:
Oh, but we do seem to use `position` (and not `stream.position`) later down...

Reply:
1. We don't need to pass `position` when append mode
2. The line below this code detects whether `position` is `undefined` to detect it's in seek mode, so doing this will confuse it

My previous commit https://github.com/kripken/emscripten/pull/6008/commits/3a95bbf7fa60f103b59706ac1e377b5f380908a2 added this because `FS.writeFile` incorrectly called `FS.write` with 0-value position.

Reply:
My previous comment was wrong, I was thinking about another part of the code. Sorry 😱